TemplateOperations/Difference/TemplateDifference.wl

```diff
@@ -0,0 +1,79 @@
+(* ::Package:: *)
+
+BeginPackage[
+  "CATemplates`TemplateOperations`Difference`TemplateDifference`",
+  {
+    "CATemplates`Basic`",
+    "CATemplates`CATemplate`",
+    "CATemplates`TemplateOperations`Expansion`PostExpansionFn`IdentityFn`",
+    "CATemplates`TemplateOperations`Expansion`PostExpansionFn`FilterNotAllowed`",
+    "CATemplates`TemplateOperations`Expansion`PostExpansionFn`FilterOutOfRange`",
+    "CATemplates`TemplateOperations`Difference`Common`",
+    "CATemplates`TemplateOperations`Difference`RawDifference`",
+    "CATemplates`TemplateOperations`Difference`ExceptionTemplates`"}
+];
+
+
+TemplateDifference::usage = "bl1";
+TemplateDifferenceDirect::usage = "bl1";
+ExpansionFunctionSelected::usage = "bl1";
+
+Begin["`Private`"];
+
+ExpansionFunctionSelected[x_, y_] := Switch[{x, y},
+  (*{IdentityFn, IdentityFn}, FilterOutOfRange,*)
+  {IdentityFn, FilterOutOfRange}, FilterOutOfRange,
+  _, x
```

Comment:
You can use pattern matching instead of a `Switch`. Check out how this is done in `TemplateIntersection`